mcs/class/System.Net.Http/corefx/PlatformDetection.cs

```diff
@@ -0,0 +1,15 @@
+namespace System
```

Comment:
Don't create a new file but use https://github.com/mono/mono/blob/master/mcs/class/test-helpers/PlatformDetection.cs instead

Reply:
No, this is intentionally in a new file by design.

It contains a few definitions that I definitively do not want to see outside the very limited context of the `System.Net.Http` tests.  This includes especially `OpenSslVersion`, `SupportsX509Chain` and `SupportsCertRevocation` (which in turn uses `UsingBtls`).

I want to keep tight control over these and - to facilitate a potential future removal / replacement with a more robust mechanism - prevent them from spreading all over the place.  This is best achieved by keeping them in a separate file.

I added a comment to the file.

Reply:
PlatformDetection is used for xunit tests only which are fixed at this point. I think it's more useful to have them in one place than spread everywhere.